### ISSUE TITLE
Make Rules work better in graphql

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#19508]
- url maps do not seem to work - either when firing on every request or when you add them to the detail page.
- Setting the persona in the query does not seem to work, I still get persona=null
- Also, it does not seem like we are accruing tags?  A visitor should accrue tags when they hit a detail page with tags in it.

```
{
  page(url: "/store/products/eagles-nest-outfitters-doublenest-hammock-patriot", fireRules:true, persona:"EcoEnthusiast") {
    viewAs{
      visitor{
        persona{
          keyTag
        }
        geo{
          continent
          country
          latitude
          longitude
        }
        tags{
          tag
        }
      }
    }
  }  
}
```

_Originally posted by @wezell in https://github.com/dotCMS/core/issues/19219#issuecomment-703885379_

**Comment by victoralfaro-dotcms** [comment]
Created rules for /store/products/eagles-nest-outfitters-doublenest-hammock-patriot
![image](https://user-images.githubusercontent.com/64026161/104385762-6f8c7d00-54f9-11eb-9606-8d052c9023e8.png)
Results using Insomnia
```
{
  "data": {
    "page": {
      "viewAs": {
        "visitor": {
          "persona": {
            "keyTag": "EcoEnthusiast"
          },
          "geo": null,
          "tags": [
            {
              "tag": "climbing"
            },
            {
              "tag": "hiking"
            },
            {
              "tag": "zip-lining"
            },
            {
              "tag": "beach"
            }
          ]
        }
      }
    }
  }
}
```
Adding count to query
```
  "data": {
    "page": {
      "viewAs": {
        "visitor": {
          "persona": {
            "keyTag": "EcoEnthusiast"
          },
          "geo": null,
          "tags": [
            {
              "tag": "climbing",
              "count": 12
            },
            {
              "tag": "hiking",
              "count": 12
            },
            {
              "tag": "zip-lining",
              "count": 12
            },
            {
              "tag": "beach",
              "count": 4
            }
          ]
        }
      }
    }
  }
```

**Comment by dsilvam** [comment]
Let's Internal QA the following: 
Now the `page` graphql field takes an optional `site` argument, which accepts either a site identifier or name. 
There's postman test for it.

**Comment by dsilvam** [comment]
**Doc:** The GraphQL Page API root field `page` takes now a new optional `site` argument to force an specific SIte when retrieving a page. The `site` arguments accept both the `identifier` or the `Host name` of the site.

**Comment by bryanboza** [comment]
Fixed, tested on release-21.03 and it is working as expected, also the postman test is running without problems
